flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertexTest.java

```diff
@@ -93,6 +92,63 @@ public void testLazyInitialization() throws Exception {
         assertThat(ejv.getTaskVertices().length, is(3));
         assertThat(ejv.getInputs().size(), is(0));
         assertThat(ejv.getProducedDataSets().length, is(1));
+        assertThat(ejv.getOperatorCoordinators().size(), is(0));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testErrorIfInitializationWithoutParallelismDecided() throws Exception {
+        final ExecutionJobVertex ejv = createDynamicExecutionJobVertex();
+
+        initializeVertex(ejv);
+    }
+
+    @Test
+    public void testSetParallelismLazily() throws Exception {
+        final int parallelism = 3;
+        final int defaultMaxParallelism = 13;
+        final ExecutionJobVertex ejv =
+                createDynamicExecutionJobVertex(-1, -1, defaultMaxParallelism);
+
+        assertThat(ejv.isParallelismDecided(), is(false));
+
+        ejv.setParallelism(parallelism);
+
+        assertThat(ejv.isParallelismDecided(), is(true));
+        assertThat(ejv.getParallelism(), is(parallelism));
+
+        initializeVertex(ejv);
+
+        assertThat(ejv.getTaskVertices().length, is(parallelism));
+    }
+
+    @Test
+    public void testConfiguredMaxParallelismIsRespected() throws Exception {
+        final int configuredMaxParallelism = 12;
+        final int defaultMaxParallelism = 13;
+        final ExecutionJobVertex ejv =
+                createDynamicExecutionJobVertex(
+                        -1, configuredMaxParallelism, defaultMaxParallelism);
+
+        assertThat(ejv.getMaxParallelism(), is(configuredMaxParallelism));
+    }
+
+    @Test
+    public void testComputingMaxParallelismFromConfiguredParallelism() throws Exception {
+        final int parallelism = 300;
+        final int defaultMaxParallelism = 13;
+        final ExecutionJobVertex ejv =
+                createDynamicExecutionJobVertex(parallelism, -1, defaultMaxParallelism);
+
+        assertThat(ejv.getMaxParallelism(), is(512));
+    }
+
+    @Test
+    public void testFallingBackToDefaultParallelism() throws Exception {
```

Comment:
should be testMaxParallelismFallingBackToDefaultMaxParallelism？

Reply:
fixed